db/migrate/20230602185438_create_publishing_pipeline_states.rb

```diff
@@ -0,0 +1,22 @@
+class CreatePublishingPipelineStates < ActiveRecord::Migration[7.0]
+  def change
+    create_table :publishing_pipeline_states do |t|
+      t.references :podcast, null: false, foreign_key: true
+      t.references :publishing_queue_item, null: false, foreign_key: true
+      t.integer :status, null: false, default: 0
+
+      t.datetime :created_at, null: false
+    end
+
+    add_index :publishing_pipeline_states,
+      [:podcast_id, :publishing_queue_item_id, :status],
+      name: "index_state_on_podcast_queue_item_and_status"
+
+    add_index :publishing_pipeline_states,
+      [:podcast_id, :publishing_queue_item_id, :status],
+      unique: true,
+      name: "index_publishing_pipeline_state_on_unique_status",
+      where: "status in (#{PublishingPipelineState.unique_status_codes.join(",")})"
```

Comment:
Does this uniqueness-except-for-rss/apple-publishing really need to be enforced?  Wondering "what's the worst that could happen"... if somehow a race condition inserted the status more than once.

This index also implies there's no retrying a PublishingQueueItem.  You have to kick off a new one.  Which I guess is fine - they're lightweight state tracking machines, with no other configuration about them.

Reply:
There should be only one pipeline happening at a time, for all the concurrency reasons with syncing state via non-idempotent operations etc..

So e.g. in the case of transitioning into `:created` twice for the same queue item, that means that two jobs are running at the same time. On the Apple synchronization side of things,  I'd expect e.g. Api errors from creating duplicate resources (for example) or models potentially ending up with stale state (with racing writes), which could be confusing to debug.

Reply:
Of course this index does nothing in the case if _differing_ `publishing_queue_item_id` values.  So from a database consistency point of view, that could be locked down more.  Or conversely, if you had to kick of a publish job twice, you could force it.